piet-wgsl/src/lib.rs

```diff
@@ -185,28 +183,29 @@ impl BlitPipeline {
             @vertex
             fn vs_main(@builtin(vertex_index) ix: u32) -> @builtin(position) vec4<f32> {
                 // Generate a full screen quad in NDCs
-                var vertex = vec2<f32>(-1.0, 1.0);
+                var vertex = vec2(-1.0, 1.0);
                 switch ix {
                     case 1u: {
-                        vertex = vec2<f32>(-1.0, -1.0);
+                        vertex = vec2(-1.0, -1.0);
                     }
                     case 2u, 4u: {
-                        vertex = vec2<f32>(1.0, -1.0);
+                        vertex = vec2(1.0, -1.0);
                     }
                     case 5u: {
-                        vertex = vec2<f32>(1.0, 1.0);
+                        vertex = vec2(1.0, 1.0);
                     }
                     default: {}
                 }
-                return vec4<f32>(vertex, 0.0, 1.0);
+                return vec4(vertex, 0.0, 1.0);
             }
             
             @group(0) @binding(0)
             var fine_output: texture_2d<f32>;
             
             @fragment
             fn fs_main(@builtin(position) pos: vec4<f32>) -> @location(0) vec4<f32> {
-                return textureLoad(fine_output, vec2<i32>(pos.xy), 0);
+                let rgba_sep = textureLoad(fine_output, vec2<i32>(pos.xy), 0);
+                return vec4(rgba_sep.rgb * rgba_sep.a, 1.0);
```

Comment:
Shouldn't the last component be rgba_sep.a instead of 1.0? This is effectively compositing over opaque black, and we should be able to generate semi-transparent windows.

Reply:
Yes, I hadn't considered semi-transparent windows. I'll change this to emit the alpha component.